Tests/Sitemap/SitemapZgipOptionTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Presta\SitemapBundle\Test\Sitemap;
+
+use Presta\SitemapBundle\Service\Dumper;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * @author Davide Dell'Erba <info@davidedellerba.it>
+ */
+class SitemapZgipOptionTest extends WebTestCase
+{
+    /**
+     * @var ContainerInterface
+     */
+    protected static $container;
+
+    /** @var EventDispatcherInterface */
+    private $eventDispatcher;
+
+    /** @var Filesystem */
+    private $filesystem;
+
+    public function setUp()
+    {
+        if (self::$kernel) {
+            static::ensureKernelShutdown();
+        }
+        self::createClient(['debug' => false]);
+        if (self::$container === null) {
+            self::$container = self::$kernel->getContainer();
+        }
+        $this->eventDispatcher = self::$container->get('event_dispatcher');
+        $this->filesystem = self::$container->get('filesystem');
+    }
+
+    public function testSitemapWithoutGzip()
+    {
+        $this->setUp();
+        $dumper = new Dumper($this->eventDispatcher, $this->filesystem);
+        $method = new \ReflectionMethod($dumper, 'loadCurrentSitemapIndex');
```

Comment:
Ouch, I will not be able to merge tests using reflection so private method can be tested unitary.

Can't you just trigger the whole process but in an isolated context ?

Reply:
Yes, I know that test private/protected methods is not so good but i saw the same thing here: https://github.com/prestaconcept/PrestaSitemapBundle/blob/bd66cb60d84d0abdad431b14a07472eca276bc32/Tests/Sitemap/SitemapindexTest.php#L43
But, it's not a problem: I'm changing the test.

Reply:
I was pretty sure we already had that kind of test. We can consider this as legacy, to be changed.